Render ts

tbosch opened the pull request.

tbosch committed:
chore(build): add tsconfig.json to modules/angular2

This allows to use `tsc` directly on Angular2, e.g.
```
./node_modules/.bin/tsc -w -p modules/angular2/
```

tbosch committed:
refactor(render): ts’ify render api

alexeagle commented on modules/angular2/globals.ts:
```diff
@@ -20,4 +22,6 @@ interface Window {
   gc(): void;
   Reflect: any;
+  zone: any;
+  Hammer: any;
```
can we avoid any here?

alexeagle commented on modules/angular2/src/core/zone/ng_zone.ts:
```diff
@@ -19,12 +19,12 @@ export class NgZone {
-  _onTurnStart:Function;
-  _onTurnDone:Function;
-  _onErrorHandler:Function;
+  _onTurnStart: Function;
```
instead of Function could be () => void; ?

alexeagle commented on modules/angular2/src/render/api.ts:
```diff
@@ -66,7 +70,7 @@ export class ElementBinder {
 export class DirectiveBinder {
   // Index into the array of directives in the View instance
-  directiveIndex:any;
+  directiveIndex: any;
```
is this any or number?

alexeagle commented on modules/angular2/src/render/api.ts:
```diff
@@ -111,25 +123,31 @@ export class ProtoViewDto {
+  callOnAllChangesDone: boolean;
   changeDetection: string;
-  constructor({id, selector, compileChildren, events, hostListeners, hostProperties,
+  constructor({
```
we should think about how to do this without destructuring

alexeagle commented on modules/angular2/src/render/dom/compiler/compile_step.ts:
```diff
@@ -6,5 +6,6 @@ import * as compileControlModule from './compile_control';
  * Is guaranteed to be called in depth first order
  */
 export class CompileStep {
```
meant to be an interface?

alexeagle commented on modules/angular2/src/render/dom/compiler/selector.ts:
```diff
@@ -329,12 +338,12 @@ class SelectorListContext {
-  cbContext; // callback context
+  selector: CssSelector;
+  notSelector: CssSelector;
+  cbContext;  // callback context
```
type?

alexeagle commented on modules/angular2/test/render/dom/dom_testbed.js:
```diff
@@ -10,13 +10,12 @@ import {RenderViewRef, ProtoViewDto, ViewDefinition, EventDispatcher, DirectiveM
 import {el, dispatchEvent} from 'angular2/test_lib';
 
-export class TestView extends EventDispatcher {
+export class TestView /* implements EventDispatcher */ {
```
why commented?

tbosch replied:
As it is not typescript yet :-(

rkirov commented on modules/angular2/tsconfig.json:
```diff
@@ -0,0 +1,8 @@
+{
+    "compilerOptions": {
+        "target": "es5",
```
add `emitDecoratorMetadata: true`

tbosch committed:
chore(build): bump ts2dart version to 0.5.5

tbosch committed:
chore(build): don’t include `export var __esModule = true` in every file

But do it during the build process for cjs.
Right now we only need this when we transpile from ts
directly to es5. This is only the case in our
cis build, as for our browser build we only transpile
from ts to es6 via ts and then use traceur to do
the rest.
